### ISSUE TITLE
Add an option to disable page ambilight and have it be only enabled in fullscreen?

### ISSUE THREAD

**Issue by iGerman00** [WesselKroos/chrome-youtube-ambilight#9]
I have an ultrawide monitor, and watching 16:9 videos on it creates big black bars, which i do not want. I found the only working extension to be this, but it also makes the entire page glow, while i only want a fullscreen glow. Can you add an option to disable page glow and only leave fullscreen on?

**Comment by WesselKroos** [comment]
@iGerman00 I've been thinking about separating all the settings per view mode. There are 3 different view modes on the YouTube website: **Fullscreen**, **Full width** and **2/3th of the width**. But I can imagine some users don't like their settings separated.

So, would a **Settings per view mode** toggle be an suitable solution for you? 
It will look like this:
(But I will probably switch this toggle with the **Enabled** setting to make clear that it also includes the **Enabled** setting.)
![YouTube Ambilight Settings per view mode](https://user-images.githubusercontent.com/31220528/71550435-58531e00-29d0-11ea-9c83-054fa2f9219e.png)

**Comment by iGerman00** [comment]
> @iGerman00 I've been thinking about separating all the settings per view mode. There are 3 different view modes on the YouTube website: **Fullscreen**, **Full width** and **2/3th of the width**. But I can imagine some users don't like their settings separated.
> 
> So, would a **Settings per view mode** toggle be an suitable solution for you?
> It will look like this:
> (But I will probably switch this toggle with the **Enabled** setting to make clear that it also includes the **Enabled** setting.)
> ![YouTube Ambilight Settings per view mode](https://user-images.githubusercontent.com/31220528/71550435-58531e00-29d0-11ea-9c83-054fa2f9219e.png)

Yes, this should do it. I have also noticed, in the recent times I have completely disabled the extension, because it seems to have a memory leak or something along the lines, and in some time it overflows the memory and the page crashes. I am using Opera

**Comment by WesselKroos** [comment]
> I have also noticed, in the recent times I have completely disabled the extension, because it seems to have a memory leak or something along the lines, and in some time it overflows the memory and the page crashes. I am using Opera
> 
> EDIT: Nevermind, I think it is fixed now? I just haven't had a chance to check.
> EDIT2: Nope, still there, the ram usage jumped to like 5.5gb for opera

 I'm aware of a memory leak in the GPU memory on Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=1015729
But I'm not aware with a RAM memory leak.

@iGerman00 Is it the memory in the red box that's increasing?
![image](https://user-images.githubusercontent.com/31220528/71560950-3ad59100-2a70-11ea-9d92-d12e3e065375.png)

**Comment by iGerman00** [comment]
> > I have also noticed, in the recent times I have completely disabled the extension, because it seems to have a memory leak or something along the lines, and in some time it overflows the memory and the page crashes. I am using Opera
> > EDIT: Nevermind, I think it is fixed now? I just haven't had a chance to check.
> > EDIT2: Nope, still there, the ram usage jumped to like 5.5gb for opera
> 
> I'm aware of a memory leak in the GPU memory on Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=1015729
> But I'm not aware with a RAM memory leak.
> 
> @iGerman00 Is it the memory in the red box that's increasing?
> ![image](https://user-images.githubusercontent.com/31220528/71560950-3ad59100-2a70-11ea-9d92-d12e3e065375.png)

Nope. It is what it is, it's RAM, not gpu memory. It then kills the page when it runs out after a couple of minutes

**Comment by WesselKroos** [comment]
Hmm let's see if I can reproduce and debug it then.
How did you install the extension on Opera? Did you install the extension from the Chrome Webstore or via a other way?

**Comment by iGerman00** [comment]
> Hmm let's see if I can reproduce and debug it then.
> How did you install the extension on Opera? Did you install the extension from the Chrome Webstore or via a other way?

I installed it from the Chrome Webstore, through the opera extension, self explanatory named “Install Chrome Extensions” on the opera add on catalog thing. Opera is Chromium after all these days.

**Comment by WesselKroos** [comment]
> I installed it from the Chrome Webstore, through the opera extension, self explanatory named “Install Chrome Extensions” on the opera add on catalog thing. Opera is Chromium after all these days.

I can't seem to reproduce the bug on Opera version 65.0.3467.78 with for example this video: https://www.youtube.com/watch?v=Bey4XXJAqS8
Could you share your YouTube Ambilight extension settings?
But maybe there is something else different between our setup.

**Comment by iGerman00** [comment]
> > I installed it from the Chrome Webstore, through the opera extension, self explanatory named “Install Chrome Extensions” on the opera add on catalog thing. Opera is Chromium after all these days.
> 
> I can't seem to reproduce the bug on **Windows 10 Version 10.0.18363 Build 18363** and **Opera version 65.0.3467.78** with for example this video: https://www.youtube.com/watch?v=Bey4XXJAqS8
> Could you share your YouTube Ambilight extension settings?
> But maybe there is something else different between our setup.

I am using [these settings](https://i.imgur.com/WkKbQvw.png), running Windows 10 1909 (Build 18363.535), and I am using Opera **_GX_**, version "LVL 1 (core: 64.0.3417.150)", it is up to date in the settings.

**Comment by WesselKroos** [comment]
@iGerman00 I can confirm the growing memory bug on Opera and created a separate issue: https://github.com/WesselKroos/chrome-youtube-ambilight/issues/14

**Comment by WesselKroos** [comment]
WIP
- [x] Detect and save view modes
- [ ] Prefix settings with view modes
- [ ] Switch between settings when changing view mode

**Comment by WesselKroos** [comment]
Moving this to Release 2.34 because switching between settings requires more rework than I thought.

**Comment by WesselKroos** [comment]
Changed the plan from creating a "Settings per view mode" setting to:
Create an option per setting to differentiate it's value per view mode.
(Excluding the settings "Advanced" and "Switch to light theme when turned off")

Since most user will probably only change one or a few options per view mode. This will prevent the hassle of having to edit every setting in each view mode when you only want to change one.
